### PR TITLE
HTML representation inside Jupyter notebook

### PR DESCRIPTION
**Pull request by lang-m**

### CODE REVIEW

**Comment by lang-m** [comment]
grafik

**Comment by marijanbeg** [comment]
Very nice!

**Comment by lang-m** [comment]
Fourier-transformed field does contain real-space mesh information (in `_repr_html_`):
grafik

**Comment by lang-m** [comment]
@marijanbeg We do currently have tests for `__repr__`. We should think about how to test the new functionality, i.e. if we write some sort of unit test for both `__str__` and `_repr_html_` or if we test the HTML representation only inside a Jupyter notebook (if this works properly).

**Comment by marijanbeg** [comment]
I think we should call these methods explicitly in tests at least. This way we ensure the coverage does not go down.

**Comment by lang-m** [comment]
@fangohr Please ignore changes in the notebooks (these almost exclusively related to rerunning the notebooks to reflect changes in representation).

File: discretisedfield/html/__init__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def strip_tags(string):
+    """Strip all html tags and convert lists."""
+    string = re.sub(r'\s+', '', string)
+    string = re.sub(r'<ul>', '(', string)
+    string = re.sub(r'<li>', '', string)
+    string = re.sub(r'</li></ul>', ')', string)
+    string = re.sub(r'</li>', ',', string)
+    string = re.sub(r'</?i>', '`', string)
```
Comment (fangohr) [inline discretisedfield/html/__init__.py:20]: What cases does `</?i>` cover?

File: discretisedfield/mesh.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import k3d
```
Comment (fangohr) [inline discretisedfield/mesh.py:1]: imports could be sorted better: internal packages first, then external (such as k3d), then our own). Within each category, ideally in alphabetical order.

`isort` will check/correct this I think.

File: MANIFEST.in [hunk 3]
```diff
@@ -2,3 +2,4 @@ include README.md
 include LICENSE
 include discretisedfield/util/plotting-style.mplstyle
 recursive-include discretisedfield/tests/test_sample *
+include discretisedfield/html/templates/*.jinja2
```
Comment (fangohr) [inline MANIFEST.in:5]: No newline at end of file. Not sure what the convention is, but github is flagging this ;)

**Comment by fangohr** [comment]
Thank you very much - this looks great. 

I added very minor suggestions. To deal with those, running `black`, `isort`  (and just to complete the trio) `flake8` on the files may be useful. Otherwise, this is all great and fine with me - go ahead and merge you you are ready.

[The template approach is great.]

**Review by fangohr — APPROVED** [review]

File: discretisedfield/html/__init__.py (hunk 1 above)
Comment (lang-m) [reply discretisedfield/html/__init__.py:20]: Names of subregions are italic [`<i>...</i>`] in the html representation.

File: discretisedfield/mesh.py (hunk 2 above)
Comment (lang-m) [reply discretisedfield/mesh.py:1]: Will be done in a separate PR

**Pull request merged**